ui/lib/musicbox/player.ex

```diff
@@ -246,6 +255,41 @@ defmodule Musicbox.Player do
       {:ok, song_list} = MpdClient.Playlists.list(item["playlist"])
       Enum.member?(song_list, path)
     end)
-    |> Enum.map(fn playlist -> playlist["playlist"] end)
+    |> Enum.map(fn playlist -> maybe_display_name(playlist["playlist"]) end)
+  end
+
+  defp set_playlist_name(id, name) do
+    new_name = case String.starts_with?(id, "#") do
+      true -> rename_changed_playlist(id, name)
+      false -> new_playlist_name(id, name)
+    end
+
+    {id, new_name}
+  end
+
+  defp rename_changed_playlist(old_name, new_name) do
+    [id | _] = String.split(old_name)
+
+    id
+    |> String.slice(1..-1)
+    |> new_playlist_name(new_name)
+  end
+
+  defp new_playlist_name(id, name) do
+    "#" <> id <> " - " <> name
+  end
+
+  defp maybe_display_name(name) do
+    case String.starts_with?(name, "#") do
+      true -> extract_playlist_name(name)
+      _ -> name
+    end
+  end
+
+  defp extract_playlist_name(name) do
```

Comment:
How about something like this where you can extract the name and ID in one go? Then we don't need any temporary indicators like `#` in strings. We can just extract the ID and name, then generate the new names straight away with `"#{id} - #{name}"`

```elixir
re = ~r/\A(?<id>\d+)(?:\W+(?<name>.+))?\z/

iex> Regex.named_captures(re, "123456 - dope tunes")
%{"id" => "123456", "name" => "dope tunes"}

iex> Regex.named_captures(re, "1234")
%{"id" => "1234", "name" => ""}

iex> Regex.named_captures(re, "derp")
nil
```